rust/candid/src/de.rs

```diff
@@ -714,6 +732,12 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
         V: Visitor<'de>,
     {
         self.unroll_type()?;
+        if matches!(
+            self.wire_type.as_ref(),
+            TypeInner::Null | TypeInner::Reserved | TypeInner::Opt(_)
+        ) {
+            self.dec_value_cost(1)?;
```

Comment:
Why do we have an if around here?

Reply:
We are only counting wire types. Deserialization can reach this function when the expected type is opt, regardless of the wire type.